### PR TITLE
Attempt to fix calendar

### PR DESCRIPTION
**Pull request by joelsa**

### CODE REVIEW

**Comment by rleh** [comment]
Viel Spaß die ICS-Generierung zu fixen:

https://github.com/TechAachen/Webseite-Jekyll/blob/6b5f0ef64d206c0b3ba05a29d8679d7d76db832e/static/calendar.ics#L34-L37

**Comment by joelsa** [comment]
Here you go:

image

[techaachen-calendar_ics.zip](https://github.com/TechAachen/Webseite-Jekyll/files/12568344/techaachen-calendar_ics.zip)

What made this at least somewhat bearable is the fact that Liquid has a Date filter: https://shopify.github.io/liquid/filters/date/ that uses Ruby's ```Time.Parse``` internally and can convert it to [strftime](https://strftime.net/) formats.

2023-11-31 parses to 2023-12-01 and 2023-02-31 parses to 2023-03-03.

You can try this out by entering
```liquid
 {{ "2023-02-31" | date: "%Y-%m-%d"}}
```
Here: https://jumpseller.in/support/liquid-sandbox/

The code that allows every month to have 31 days and then wraps around in Ruby is located here:
https://github.com/ruby/time/blob/7bcbedf207085d4a11954181a89306a8c15d3393/lib/time.rb#L219

By using this I don´t have to handle different month durations or leap years. This code also works for events that take place across different years (e.g. over new years eve).

This may fail for any event lasting longer than 28 days, so 4 weeks, or 672 hours is the maximum possible duration.

File: _data/calendar.yml [hunk 1]
```diff
@@ -615,10 +615,5 @@
 - title: TechTurbo 2023
   datum: "2023-11-30"
   uhrzeit: "16.00h"
-  dauer: 8
-  ort: "Gruppenunterkunft Auszeit Eifel, Auf der Batterie 9, Schleiden"
-- title: TechTurbo 2023
-  datum: "2023-12-01"
-  uhrzeit: "00.00h"
-  dauer: 62
+  dauer: 70
```
Comment (rleh) [inline _data/calendar.yml:618]:
```suggestion
  dauer: 72
```

File: static/calendar.ics [hunk 2]
```diff
@@ -34,7 +34,36 @@ END:VTIMEZONE
 {% assign end_uhrzeit = termin.uhrzeit | remove: "h" | remove: "." | plus: dauer -%}
 {% assign dauer_tage = end_uhrzeit | divided_by: 2400 -%}
 {% assign end_uhrzeit = end_uhrzeit | modulo: 2400 -%}
-{% assign end = termin.datum | remove: "-" | plus: dauer_tage | append: "T" | append: end_uhrzeit | append: "00" -%}
+{% assign starttag = termin.datum | date: '%d' %}
+{% assign startmonat = termin.datum | date: '%m' %}
+{% assign startjahr = termin.datum | date: '%Y' %}
```
Comment (rleh) [inline static/calendar.ics:39]:
```suggestion
{% assign starttag = termin.datum | date: '%d' -%}
{% assign startmonat = termin.datum | date: '%m' -%}
{% assign startjahr = termin.datum | date: '%Y' -%}
```

**Review by rleh — COMMENTED** [review]
grafik
(`DTSTAMP` is expected to change.)

Not sure if iCal allows blank lines, but there is an easy fix:

**Review by rleh — APPROVED** [review]

**Pull request closed without merging**